### ISSUE TITLE
Rule proposal: no-magic-booleans

### ISSUE THREAD

**Issue by pedroluiznogueira** [typescript-eslint/typescript-eslint#7441]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+new+plugin+rule%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] My proposal specifically checks TypeScript syntax, or it proposes a check that requires type information to be accurate.
- [X] My proposal is not a "formatting rule"; meaning it does not just enforce how code is formatted (whitespace, brace placement, etc).
- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

The rule should identify any function or method call that directly receives a boolean literal as one of its arguments and raise a warning or error.

### Fail Cases

```typescript
setUserStatus(userId, true);
doSomething(false);
```

### Pass Cases

```typescript
const isActive = true;
setUserStatus(userId, isActive);

const shouldSkip = false;
doSomething(shouldSkip);
```

**Comment by bradzacher** [comment]
Thanks for this request!

> [x] My proposal specifically checks TypeScript syntax, or it proposes a check that requires type information to be accurate.

This request doesn't have any TS-specific checks, nor does it specifically require type-information.

Additionally this is a stylistic preference which doesn't broadly apply one way or the other to the typescript community and the style doesn't impact type quality or improve safety.

For these reasons, I'm going to reject this proposal from this project.

Feel free to build on top of our tooling and publish this as a stand-alone plugin, or submit it to an existing, general-purpose plugin!

**Comment by pedroluiznogueira** [comment]
> Thanks for this request!
> 
> > [x] My proposal specifically checks TypeScript syntax, or it proposes a check that requires type information to be accurate.
> 
> This request doesn't have any TS-specific checks, nor does it specifically require type-information.
> 
> Additionally this is a stylistic preference which doesn't broadly apply one way or the other to the typescript community and the style doesn't impact type quality or improve safety.
> 
> For these reasons, I'm going to reject this proposal from this project.
> 
> Feel free to build on top of our tooling and publish this as a stand-alone plugin, or submit it to an existing, general-purpose plugin!

Thanks for your response! Does anyone have any suggestions of a project/plugin where this would fit? Appreciate it…
